### ISSUE TITLE
Cannot "go get" by following the installation guide

### ISSUE THREAD

**Issue by rajat12a** [algolia/algoliasearch-client-go#470]
- Algolia Client Version: master
- Language Version: 1.10.3

### Description
Trying to follow the installation steps in my machine and get the following error

```
package github.com/algolia/algoliasearch-client-go/algoliasearch: cannot find package "github.com/algolia/algoliasearch-client-go/algoliasearch" in any of:
	/usr/local/go/src/github.com/algolia/algoliasearch-client-go/algoliasearch (from $GOROOT)
```

### Steps To Reproduce

go get github.com/algolia/algoliasearch-client-go/algoliasearch

**Comment by kurtsson** [comment]
I re-released v2.27.0 as a module temporarily here: https://github.com/daresaydigital/algoliasearch-client-go.
It fixes the problem for now

**Comment by rajat12a** [comment]
@kurtsson This works, but shouldn't algolia keep a support for v2.x

A sudden breaking change without any prior warning/notice
Also is there any way to use `go get` for a specific version?

**Comment by kurtsson** [comment]
> @kurtsson This works, but shouldn't algolia keep a support for v2.x
> 
> A sudden breaking change without any prior warning/notice
> Also is there any way to use `go get` for a specific version?

Yes, I didn't suggest it was a proper "fix". Just a temporary solution until Algolia fixes their stuff

**Comment by aseure** [comment]
Hello @rajat12a. Thank you for raising this issue.

We are in the process of releasing a new major version of our Go client. Because this is still work in progress, we've simply released a beta version of it (`v3.0.0-beta01`). Starting with this version will support both `dep` as a dependency manager and the newly introduced Go modules. However, we cannot make changes to previous versions to support Go modules. The only way to fetch those previous versions is to us `dep` as it was the case before.

The issue you are getting is due to `go get` trying to fetch a subpackage which doesn't exist anymore on `master`. By using a package manager (like `dep`) or Go modules you should not have any issue in the future. I'm sorry for the miscommunication around this beta release. I was expecting everyone to use the previously mentioned solutions and forgot about this (wrong) default notice of installation we had in our README.md. Anyway, it's now changed as of https://github.com/algolia/algoliasearch-client-go/commit/fca2d2fc84d8242e2c573318816fa7a4eba22eee, thanks for letting me know.

@kurtsson Thank you to you as well for taking the time to help and for the workaround. As said, I've now updated the README.md to show-case the different correct ways to install the Go client.

Should you have any other issue with the installation process, please let us know by reopening this issue. Thank you for taking the time to share your feedbacks. Migration guides and documentation are coming soon for the v3 and we've managed to integrate a lot of new features. Stay tuned!

**Comment by aseure** [comment]
Hello everyone,

Extra note: I've created a `v2` branch to the repository. This way, the v2 version is still `go get`-able using the following command:

```
# Use:
go get github.com/algolia/algoliasearch-client-go@v2

# instead of:
go get github.com/algolia/algoliasearch-client-go/algoliasearch
```

README has been updated accordingly https://github.com/algolia/algoliasearch-client-go/commit/7a87f8eb3d7259dc85ea72bf239fb0681f555838

cc @rajat12a @kurtsson
